internal/cli/login.go

```diff
@@ -0,0 +1,141 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	"github.com/flightctl/flightctl/api/v1alpha1"
+	apiClient "github.com/flightctl/flightctl/internal/api/client"
+	"github.com/flightctl/flightctl/internal/client"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+type LoginOptions struct {
+	GlobalOptions
+	Token string
+}
+
+func DefaultLoginOptions() *LoginOptions {
+	return &LoginOptions{
+		GlobalOptions: DefaultGlobalOptions(),
+		Token:         "",
+	}
+}
+
+func NewCmdLogin() *cobra.Command {
+	o := DefaultLoginOptions()
+	cmd := &cobra.Command{
+		Use:   "login [URL] --token TOKEN",
+		Short: "Login to flight control",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := o.Complete(cmd, args); err != nil {
+				return err
+			}
+			if err := o.Validate(args); err != nil {
+				return err
+			}
+			return o.Run(cmd.Context(), args)
+		},
+		SilenceUsage: true,
+	}
+
+	o.Bind(cmd.Flags())
+	return cmd
+}
+
+func (o *LoginOptions) Bind(fs *pflag.FlagSet) {
+	o.GlobalOptions.Bind(fs)
+
+	fs.StringVarP(&o.Token, "token", "t", o.Token, "Bearer token for authentication to the API server")
+}
+
+func (o *LoginOptions) Complete(cmd *cobra.Command, args []string) error {
+	if err := o.GlobalOptions.Complete(cmd, args); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *LoginOptions) Validate(args []string) error {
+	if err := o.GlobalOptions.Validate(args); err != nil {
+		return err
+	}
+	return nil
+}
+
+type OauthServerResponse struct {
+	TokenEndpoint string `json:"token_endpoint"`
+}
+
+func (o *LoginOptions) Run(ctx context.Context, args []string) error {
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return err
+	}
+	configFilePath := filepath.Join(homeDir, ".flightctl", "client.yaml")
+	config, err := client.ParseConfigFile(configFilePath)
+	if err != nil {
+		return err
+	}
+
+	config.Service.Server = args[0]
```

Comment:
Totally fine to leave this for a follow up PR:
When the user needs to log in a second time (because the token expired), the user wouldn't need to specify the server as argument anymore but we could take the server URL from the client.yaml directly, or? In that case, wondering whether the test would belong here or in Validate().